console_win.go

```diff
@@ -496,79 +513,95 @@ func mrec2btns(mbtns, flags uint32) ButtonMask {
 }
 
 func (s *cScreen) getConsoleInput() error {
-	rec := &inputRecord{}
-	var nrec int32
-	rv, _, er := procReadConsoleInput.Call(
-		uintptr(s.in),
-		uintptr(unsafe.Pointer(rec)),
-		uintptr(1),
-		uintptr(unsafe.Pointer(&nrec)))
-	if rv == 0 {
-		return er
-	}
-	if nrec != 1 {
-		return nil
-	}
-	switch rec.typ {
-	case keyEvent:
-		krec := &keyRecord{}
-		krec.isdown = geti32(rec.data[0:])
-		krec.repeat = getu16(rec.data[4:])
-		krec.kcode = getu16(rec.data[6:])
-		krec.scode = getu16(rec.data[8:])
-		krec.ch = getu16(rec.data[10:])
-		krec.mod = getu32(rec.data[12:])
-
-		if krec.isdown == 0 || krec.repeat < 1 {
-			// its a key release event, ignore it
+	waitObjects := []syscall.Handle{s.in, s.cancelflag}
+
+	rv, _, er := procWaitForMultipleObjects.Call(
+		uintptr(len(waitObjects)),
+		uintptr(unsafe.Pointer(&waitObjects[0])),
+		uintptr(0),
+		^uintptr(0))
+	switch rv {
+	case 0: // s.in
```

Comment:
So technically, the win32 API doesn't call these 0, 1, etc.  Instead they say relative to WAIT_OBJECT_0.  I presume you've verified that WAIT_OBJECT_0 is actually *0*?  Again, I'm not a Win32 expert here, so I just use the Microsoft API documentation.  Some commentary here might be helpful.